lib/ansible/module_utils/facts.py

```diff
@@ -159,6 +159,9 @@ class Facts(object):
                  { 'path' : '/usr/local/sbin/pkg',  'name' : 'pkgng' },
     ]
 
+    # Allowed fact subset for gather_subset options
+    ALLOWED_FACT_SUBSET = frozenset([ 'all', 'min', 'network', 'hardware', 'virtual' ])
```

Comment:
I just noticed that this is a class attribute of Facts but it isn't used in the Facts class -- only in the ansible_facts() function.  Because of that, just make this a global value in this file.  No need to make it part of the class.